pkg/cmd/provisioning/identity_provider.go

```diff
@@ -314,23 +312,21 @@ func createOIDCBucket(client aws.Client, bucketName, namePrefix, region, targetD
 				return errors.Wrap(err, "Failed to create a bucket to store OpenID Connect configuration")
 			}
 		} else {
-			bucketTagValue = ownedCcoctlAWSResourceTagValue
 			log.Print("Bucket ", bucketName, " created")
-		}
-
-		_, err = client.PutBucketTagging(&s3.PutBucketTaggingInput{
-			Bucket: awssdk.String(bucketName),
-			Tagging: &s3.Tagging{
-				TagSet: []*s3.Tag{
-					{
-						Key:   awssdk.String(fmt.Sprintf("%s/%s", ccoctlAWSResourceTagKeyPrefix, namePrefix)),
-						Value: awssdk.String(bucketTagValue),
+			_, err = client.PutBucketTagging(&s3.PutBucketTaggingInput{
+				Bucket: awssdk.String(bucketName),
+				Tagging: &s3.Tagging{
+					TagSet: []*s3.Tag{
+						{
+							Key:   awssdk.String(fmt.Sprintf("%s/%s", ccoctlAWSResourceTagKeyPrefix, namePrefix)),
+							Value: awssdk.String(ownedCcoctlAWSResourceTagValue),
+						},
```

Comment:
@joelddiaz This will also tag preexisting buckets as `owned`. How should we handle a scenario where in a user has created a bucket with other objects (other then STS)? We should not allow `ccoctl delete` to remove that bucket right? This was my thinking when tagging preexisting buckets as `shared`. If bucket is shared `ccoctl delete` should only remove the tag and sts objects and not delete the entire bucket.

Reply:
the code lives under the `else` which should only run when ccoctl actually created the bucket.

Reply:
Oops my bad. Looks good